plugins/flowdock/lib/samson_flowdock/flowdock_service.rb

```diff
@@ -0,0 +1,64 @@
+require 'flowdock'
+
+module SamsonFlowdock
+  class FlowdockService
+
+    def initialize(deploy)
+      @deploy = deploy
+    end
+
+    def users
+      flowdock_client.get('/users').map do |user|
+        {
+          id: user['id'],
+          name: user['nick'],
+          avatar: user['avatar'],
+          type: user['contact']
+        }
+      end
+    end
+
+    def flowdock_client
+      @flowdock_client ||= Flowdock::Client.new(api_token: ENV['FLOWDOCK_API_TOKEN'])
+    end
+
+    def notify_chat(message, tags)
+      chat_flow.push_to_chat(content: message, tags: tags)
+    end
+
+    def notify_inbox(subject, message)
+      tags = ["deploy", stage.name.downcase]
+      inbox_flow.push_to_team_inbox(subject: subject, content: message, tags: tags, link: @deploy.deploy_url)
+    end
+
+    private
+
+    def inbox_flow
+      @flow ||= Flowdock::Flow.new(
+        api_token: tokens,
+        source: "samson",
+        from: { name: user.name, address: user.email }
+      )
+    end
+
+    def project
+      @project ||= stage.project
+    end
+
+    def stage
+      @stage ||= @deploy.stage
+    end
+
+    def user
+      @user ||= @deploy.user
+    end
```

Comment:
This seems needless to create these functions to save you typing ```deploy```, but I'm not willing to argue over it! :)